libvcx/src/v3/utils/encryption_envelope.rs

```diff
@@ -62,20 +64,24 @@ impl EncryptionEnvelope {
     }
 
     pub fn open(payload: Vec<u8>) -> VcxResult<A2AMessage> {
-        let unpacked_msg = crypto::unpack_message(&payload)?;
+        trace!("EncryptionEnvelope::open >>> payload: {:?}", payload);
 
-        let message: ::serde_json::Value = ::serde_json::from_slice(unpacked_msg.as_slice())
-            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize message: {}", err)))?;
+        let message = if AgencyMockDecrypted::has_decrypted_mock_messages() {
```

Comment:
I think I'd rather be in favor of having separate "Mock container" for usage in this place. Currently it might be harder to understand  where the data from `AgencyMockDecrypted ` will be really used when I set it up in test. I'd suggest something like `ConnectionMessageMockDecrypted`. Leaving to your judgement, just consider this idea.

Reply:
Oh I see... this `get_next_decrypted_message` is separate method from `get_next_decrypted_response`. Then nevermind! This is good